### PR TITLE
simplify get/delete for scalesets

### PR DESCRIPTION
**Pull request by bmc-msft**

### CODE REVIEW

**Review by chkeita — APPROVED** [review]

File: src/api-service/__app__/onefuzzlib/azure/vmss.py [hunk 1]
```diff
@@ -46,16 +46,10 @@ def list_vmss(name: UUID) -> Optional[List[str]]:
...
-    except ResourceNotFoundError:
-        return True
-    except CloudError as err:
-        logging.error("cloud error deleting vmss: %s (%s)", name, err)
-
-    return False
+    response = compute_client.virtual_machine_scale_sets.begin_delete(
+        resource_group, str(name)
+    )
+    return bool(response.status() == "Succeeded")
```
Comment (ranweiler) [inline src/api-service/__app__/onefuzzlib/azure/vmss.py:52]: I suggest documenting the source of the `"Succeeded"` string, assuming there are no enums in play that mypy is leveraging.

File: src/api-service/__app__/onefuzzlib/orm.py [hunk 2]
```diff
@@ -85,6 +86,11 @@ def process_state_update(obj: HasState) -> None:
     func = getattr(obj, obj.state.name, None)
     if func is None:
         return
+
+    get_keys = getattr(obj, "get_keys", None)
+    if get_keys is not None:
+        logging.info("processing state update: %s - %s", get_keys(), obj.state.name)
```
Comment (ranweiler) [inline src/api-service/__app__/onefuzzlib/orm.py:92]: This is just for extra debuggability, right?

Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/orm.py:92]: Yep.

File: src/api-service/__app__/onefuzzlib/azure/vmss.py (hunk 1 above)
Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/azure/vmss.py:52]: I can't find the documented value.  The API returns a poller of type None.  It's... not pleasant.

Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/azure/vmss.py:59]: added comments

**Pull request merged**